app/src/main/java/com/futurice/freesound/feature/common/ui/adapter/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,104 @@
+package com.futurice.freesound.feature.common.ui.adapter;
+
+import com.futurice.freesound.common.utils.AndroidPreconditions;
+import com.futurice.freesound.feature.common.DisplayableItem;
+import com.futurice.freesound.inject.fragment.FragmentScope;
+
+import android.support.annotation.NonNull;
+import android.support.v7.util.DiffUtil;
+import android.support.v7.widget.RecyclerView;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+import javax.inject.Inject;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+/**
+ * Implementation of {@link android.support.v7.widget.RecyclerView.Adapter} for {@link
+ * DisplayableItem}.
+ */
+@FragmentScope
+public final class RecyclerViewAdapter extends RecyclerView.Adapter {
```

Comment:
I'm a little bit skeptical about marking this as `@FragmentScope`. Generally speaking, we shouldn't enforcing this here, but in the usage as required by the application - so leave this to the Dagger module (this is again based upon the idea that this adapter logic should not depend on how it is used in this particular application).